modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -789,13 +789,7 @@ private RetryPolicyTemplate getRetryPolicyTemplate(Optional<ConfigurationInstanc
   }
 
   private RetryPolicyTemplate fetchRetryPolicyTemplate(Optional<ConfigurationInstance> configuration) {
-    RetryPolicyTemplate delegate = null;
-    if (retryPolicyTemplate != null) {
-      delegate = configuration
-          .map(config -> config.getConnectionProvider().orElse(null))
-          .map(provider -> connectionManager.getReconnectionConfigFor(provider).getRetryPolicyTemplate(retryPolicyTemplate))
-          .orElse(retryPolicyTemplate);
-    }
+    RetryPolicyTemplate delegate = retryPolicyTemplate;
```

Comment:
if delegate is null and the one obtained from the connection provider is async, should it be unwrapped?

Reply:
In that case it behaves the same way as before. It implies that there no reconnection config overriding in the op so this will be resolved from the connection manager.

Reply:
I've changed so this is unwrapped now.